test/suites/telemetry/telemetry.robot

```diff
@@ -100,22 +97,16 @@ MicroShift Fails to Report Metrics To Default Server With Wrong Pull Secret
 
 MicroShift Reports Metrics To Prometheus Server
     [Documentation]    Check the expected metrics are sent to the local server.
-    [Setup]    Run Keywords
-    ...    Start Prometheus Server    ${PROMETHEUS_PORT}
-    ...    AND
-    ...    Setup Telemetry Configuration    ${ENABLE_TELEMETRY_TO_PROMETHUS}    ${PULL_SECRET_METRICS}
+    [Setup]    Setup Telemetry Configuration    ${ENABLE_TELEMETRY_TO_PROMETHUS}    ${PULL_SECRET_METRICS}
 
     Should Find Metrics In Journal Log Success    Metrics sent successfully
 
     @{metrics_to_check}=    Get List Prometheus Metrics To Check
     FOR    ${metric}    IN    @{metrics_to_check}
-        Check Prometheus Query    ${PROMETHEUS_HOST}    ${PROMETHEUS_PORT}    ${metric}
+        Check Prometheus Query    ${PROMETHEUS_HOST}    ${PROMETHEUS_PORT}    ${metric}    add_hostname_filter=${False}
     END
 
-    [Teardown]    Run Keywords
-    ...    Remove Telemetry Configuration
-    ...    AND
-    ...    Stop Prometheus Server    ${PROMETHEUS_PORT}
+    [Teardown]    Remove Telemetry Configuration
```

Comment:
This `MicroShift Reports Metrics To Prometheus Server` Test Case must be executed always in the last place in this Test Suite. If not, the other 2 `Prometheus` Test Cases will fail because they'll find metrics in the Prometheus server.
This is the reason why, in the original implementation, the Prometheus server starts and stops for every Test Case.